File: Solutions/20257/python3/program.py
def find_nearest_station(k: int, position: int) -> int:
    remain = abs(position) % k
    if remain == 0:
        return position
    if abs(position) % k > k / 2:
        return k * round(position / k)
    return k * int(position / k)


def main():
    k, person1, person2 = map(int, input().split())

    only_train1 = abs(person1 - person2)

    person1_nearest_station = find_nearest_station(k, person1)
    train1_and_train2 = abs(person1_nearest_station - person1)

    person2_nearest_station = find_nearest_station(k, person2)
    train1_and_train2 += abs(person2_nearest_station - person2)

    remain_distance = abs(person2_nearest_station - person1_nearest_station)
    train1_and_train2 += remain_distance // k

    print(min(train1_and_train2, only_train1))


if __name__ == '__main__':
    main()

File: Solutions/20257/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            GetInputs()
                .CalculateMinimumTime()
                .PrintLine();
        }

        private static int CalculateMinimumTime(this Inputs inputs) {
            var onlyTrain1 = Math.Abs(inputs.Position1 - inputs.Position2);

            var position1NearestStation = FindNearestStation(inputs.K, inputs.Position1);
            var train1AndTrain2 = Math.Abs(position1NearestStation - inputs.Position1);

            var position2NearestStation = FindNearestStation(inputs.K, inputs.Position2);
            train1AndTrain2 += Math.Abs(position2NearestStation - inputs.Position2);

            var remainDistance = Math.Abs(position2NearestStation - position1NearestStation) / inputs.K;
            train1AndTrain2 += remainDistance;

            return Math.Min(train1AndTrain2, onlyTrain1);
        }

        private static int FindNearestStation(int k, int position) {
            var remain = Math.Abs(position) % k;
            if (remain == 0)
                return position;
            if (Math.Abs(position) % k > k / 2)
                return (int) (k * Math.Round(position * 1.0 / k));
            return k * (position / k);
        }

        private static void PrintLine<T>(this T data) {
            Console.WriteLine(data);
        }

        private static List<T> GetInputs<T>(char separator) =>
            Console.ReadLine()
                ?.Trim()
                .Split(separator)
                .Select(item => (T) Convert.ChangeType(item, typeof(T))).ToList()
            ?? new List<T>();

        private static Inputs GetInputs() {
            var inputs = GetInputs<int>(' ');
            return new Inputs {
                K = inputs[0],
                Position1 = inputs[1],
                Position2 = inputs[2]
            };
        }
    }

    public class Inputs {
        public int K { get; set; }
        public int Position1 { get; set; }
        public int Position2 { get; set; }
    }
}